cql3/statements/select_statement.cc

```diff
@@ -914,6 +914,7 @@ indexed_table_select_statement::do_execute(service::storage_proxy& proxy,
             return repeat([this, &builder, &options, &internal_options, &proxy, &state, now, whole_partitions, partition_slices, restrictions_need_filtering] () {
                 auto consume_results = [this, &builder, &options, &internal_options, restrictions_need_filtering] (foreign_ptr<lw_shared_ptr<query::result>> results, lw_shared_ptr<query::read_command> cmd) {
                     if (restrictions_need_filtering) {
+                        _stats.filtered_rows_read_total += *results->row_count();
```

Comment:
Doesn't `_stats.filtered_rows_matched_total` need to get bumped too? All the other sites bump these two together.

Reply:
It is bumped below, in line 947 - in github you can use the blue accordion button to see more changes. The reason why they're not bumped together is that indexed queries perform an additional phase of reading base rows before the whole result is ready and `filtered_rows_matched_total` is bumped with the size of the whole result.